### PR TITLE
DOC: Fix sphinx errors and warnings

### PR DESCRIPTION
**Pull request by bashtage**
Large scale fix up of small issues so that sphinx will build
the documentation correctly.

### CODE REVIEW

File: docs/source/plots/graphics_month_plot.py [hunk 1]
```diff
@@ -4,9 +4,8 @@
 dta = sm.datasets.elnino.load_pandas().data
 dta['YEAR'] = dta.YEAR.astype(int).astype(str)
 dta = dta.set_index('YEAR').T.unstack()
-dates = map(lambda x : pd.datetools.parse_time_string('1 '+' '.join(x)),
-                                       dta.index.values)
+dates = pd.to_datetime(list(map(lambda x : '-'.join(x) + '-1', dta.index.values)))
 
-dta.index = pd.DatetimeIndex(dates, freq='M')
+dta.index = pd.DatetimeIndex(list(dates), freq='M')
 dta.name = 'temp'
```
Comment (bashtage) [inline docs/source/plots/graphics_month_plot.py]: The list is not needed here.

File: docs/sphinxext/numpy_ext/docscrape.py [hunk 2]
```diff
@@ -6,7 +6,10 @@
 import textwrap
```
Comment (josef-pkt) [inline docs/sphinxext/numpy_ext/docscrape.py]: I thought we dropped/deleted `sphinxext/numpy_ext`

File: statsmodels/duration/__init__.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+from statsmodels.duration.survfunc import SurvfuncRight
```
Comment (josef-pkt) [inline statsmodels/duration/__init__.py]: do we keep `__init__.py` empty in duration and use api instead?   I guess so.

**Comment by josef-pkt** [comment]
@bashtage  Thanks, that's very helpful

I need to check what's now with numpydoc. I thought @TomAugspurger removed it.
I think that we want to keep the `duration.__init__.py` empty and add and api.py instead for forwards compatibility when we get more duration models.

File: docs/sphinxext/numpy_ext/docscrape.py (hunk 2 above)
Comment (TomAugspurger) [inline docs/sphinxext/numpy_ext/docscrape.py]: I did remove `sphinxext/numpydoc`, in favor of installing the one from pip. Didn't touch / test what `numpy_ext` is doing.

Comment (josef-pkt) [inline docs/sphinxext/numpy_ext/docscrape.py]: I will check this. I thought numpy_ext is numpydoc

**Comment by josef-pkt** [comment]
Yes, numpy_ext is our old included version of numpydoc including a plot_directive when matplotlib didn't have one yet. So, AFAICS we can completely remove `numpy_ext`

conf.py still has
```
if LooseVersion(mpl_version) < LooseVersion('1.0.1'):
    extensions.remove('matplotlib.sphinxext.plot_directive')
    extensions.append('numpy_ext.plot_directive')
```

File: statsmodels/duration/__init__.py (hunk 3 above)
Comment (bashtage) [inline statsmodels/duration/__init__.py]: Where should this be?  statsmodels.api?

**Comment by bashtage** [comment]
> conf.py still has

This is obviously old and can be removed.

**Comment by bashtage** [comment]
> I thought numpy_ext is numpydoc
> Add a line note

Is numpy_ext deletable?

**Comment by josef-pkt** [comment]
> Is numpy_ext deletable?

yes, AFAICS 
you could delete it and check the docbuild that it doesn't have unexpected error.

**Comment by josef-pkt** [comment]
There is an extra empty file in this changeset `statsmodels/problem.py`

merge ASAP to avoid future merge conflicts.

**Comment by bashtage** [comment]
Forgot to del that file.

**Comment by josef-pkt** [comment]
Thanks, merging

**Pull request merged**